README.md

```diff
@@ -1,486 +1,679 @@
 
 ```mjs
-// bound.mjs
-export decorator @bound {
-  @register((target, name) => {
-    const method = target[name];
-    Object.defineProperty(target, name, {
-      get() {
-        const bound = method.bind(this);
-        Object.defineProperty(this, name, { value: bound, configurable: true });
-      }
-      configurable: true
-    });
-  })
+// private-key.mjs
+export class PrivateKey {
+  #get;
+  #set;
+  
+  show({get, set}) {
+    assert(this.#get === undefined && this.#set === undefined);
+    this.#get = get;
+    this.#set = set;
+    return {get, set};
+  }
+  get(obj) {
+    return this.#get(obj);
+  }
+  set(obj, value) {
+    return this.#set(obj, value);
+  }
 }
 ```
 
-There are various approaches to writing an auto-bound decorator, but ultimately, the most efficient way may be built into the JavaScript engine; see [NEXTBUILTINS.md](./NEXTBUILTINS.md#bound) for discussion of a built-in `@bound` decorator, and the [bound-decorator](https://github.com/mbrowne/bound-decorator) repository for another approach.
+This example could be roughly desugared as follows:
 
-### `@callable`
+```mjs
+let initialize, get, set;
+export class Box {
+  #_contents = initialize(undefined);
+  get #contents() { return get.call(this); }
+  set #contents(v) { set.call(this, v); }
+
+  static #_ = (
+    get = function() { return this.#_contents; },
+    set = function(v) { this.#_contents = v; }
+  )
+}
+{get, set, initialize} = key.show({get, set}, {kind: "field", isStatic: false});
+```
+
+### `@deprecated`
 
-The `@callable` decorator makes it possible to invoke the class without `new`. When a class decorated with `@callable` is called, its static `call` method is invoked.
+The `@deprecated` decorator prints warnings when a deprecated field, method or accessor is used. As an example usage:
 
 ```mjs
-import { @callable } from "./callable.mjs";
+import { deprecated } from "./deprecated.mjs"
+
+export class MyClass {
+  @deprecated field;
 
-@callable
-class MyDate {
-  static call(...args) { return Date(...args) }
-  constructor(...args) { return new Date(...args) }
+  @deprecated method() { }
+
+  otherMethod() { }
 }
 ```
 
-An implementation in terms of `@wrap`:
+To allow the `deprecated` to work on different kinds of class elements, the `kind` field of the context object lets decorators see which kind of syntactic construct they are deprecating. This technique also allows an error to be thrown when the decorator is used in a context where it can't apply--for example, the entire class cannot be marked as deprecated, since there is no way to intercept its access.
```

Comment:
why couldn't the entire class be deprecated, by replacing the constructor and every prototype method with something that logs a warning?

Reply:
Sure, you could do that. I'd be happy to take a suggestion for how to word this paragraph better.

Reply:
I think at least implementing it for classes and showing a use of it would help make it clear.